lib/zod/index.ts

```diff
@@ -0,0 +1,21 @@
+import { ZodType } from 'zod';
+import { ApiError } from '../errors';
+
+export * from './schema';
+
+export const validateWithSchema = (
+  schema: ZodType,
+  data: any
+): {
+  [key: string]: any;
+} => {
+  const result = schema.safeParse(data);
+
+  if (!result.success) {
+    throw new ApiError(
+      422,
+      `Validation Error: ${result.error.errors.map((e) => e.message)[0]}`
+    );
+  }
+  return result.data;
+};
```

Comment:
This will infer the type of the schema you passed in otherwise it will be `any`

```suggestion
export const validateWithSchema = <ZSchema extends ZodType>(
  schema: ZSchema,
  data: any
) => {
  const result = schema.safeParse(data);

  if (!result.success) {
    throw new ApiError(
      422,
      `Validation Error: ${result.error.errors.map((e) => e.message)[0]}`
    );
  }

  return result.data as z.infer<ZSchema>;
};

```